docs: removed unnecessary exports previously added to api docs

am29d opened the pull request:
## Summary

### Changes

> Please provide a summary of what's being changed

This PR removes not supported api docs annotations and reverts exports that were previously added. Instead we now use `missing-exports` plugin to collect all implicit exports in api docs.

Current api docs build looks clean:

```
> aws-lambda-powertools-typescript@0.0.1 docs-generateApiDoc
> typedoc .

[info] Loaded plugin typedoc-plugin-zod
[info] Loaded plugin typedoc-plugin-missing-exports
[info] Converting project at ./packages/commons
[info] Converting project at ./packages/jmespath
[info] Converting project at ./packages/logger
[info] Converting project at ./packages/metrics
[info] Converting project at ./packages/tracer
[info] Converting project at ./packages/parameters
[info] Converting project at ./packages/idempotency
[info] Converting project at ./packages/batch
[info] Converting project at ./packages/parser
[info] Merging converted projects
[info] Documentation generated at ./api
```

<!-- What is this PR solving? Write a clear description or reference the issue(s) it addresses. -->

> Please add the issue number below, if no issue is present the PR might get blocked and not be reviewed

**Issue number:** closes #2993 

<!-------
Before creating the pull request, please make sure you do the following:

- Read the Contributing Guidelines at https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/CONTRIBUTING.md#sending-a-pull-request
- Check that there isn't already a PR that addresses the same issue. If you find a duplicate, please leave a comment under the existing PR so we can discuss how to move forward
- Check that the change meets the project's tenets https://docs.powertools.aws.dev/lambda/typescript/latest/#tenets
- Add a PR title that follows the conventional commit semantics - https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/.github/semantic.yml#L2
- If relevant, add tests that prove that the change is effective and works
- Whenever relevant, make sure to comment functions/methods/types and make appropriate changes to the documentation
------->

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

dreamorosi reviewed (approved):
I see that we've added the `packages/testing` in the `typedoc.json` file, but I don't see it in the build API docs outputs.

I just wanted to make sure the package does not end up in the published docs.

am29d commented:
> I see that we've added the `packages/testing` in the `typedoc.json` file, but I don't see it in the build API docs outputs.
> 
> I just wanted to make sure the package does not end up in the published docs.

I have added it to the `exclude` block.
